packages/state-transition/src/cache/epochCache.ts

```diff
@@ -428,6 +441,8 @@ export class EpochCache {
       // Common append-only structures shared with all states, no need to clone
       pubkey2index: this.pubkey2index,
       index2pubkey: this.index2pubkey,
+      // Fork-aware cache needs to be cloned. But by the virtue of it being persistent, we don't need to do anything here
+      unfinalizedIndex2pubkey: this.unfinalizedPubkey2Index,
```

Comment:
You need to clone the root node such that mutations of a child copy do not affect the parent